lib/mysql.js

```diff
@@ -550,25 +550,48 @@ MySQL.prototype.ping = function(cb) {
 
 MySQL.prototype.buildExpression = function(columnName, operator, operatorValue,
   propertyDefinition) {
-  if (operator === 'regexp') {
-    let clause = columnName + ' REGEXP ?';
-    // By default, MySQL regexp is not case sensitive. (https://dev.mysql.com/doc/refman/5.7/en/regexp.html)
-    // To allow case sensitive regexp query, it has to be binded to a `BINARY` type.
-    // If ignore case is not specified, search it as case sensitive.
-    if (!operatorValue.ignoreCase) {
-      clause = columnName + ' REGEXP BINARY ?';
-    }
-
-    if (operatorValue.ignoreCase)
-      g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`i`}} flag');
-    if (operatorValue.global)
-      g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`g`}} flag');
+  let clause;
+  switch (operator) {
+    case 'regexp':
+      clause = columnName + ' REGEXP ?';
+      // By default, MySQL regexp is not case sensitive. (https://dev.mysql.com/doc/refman/5.7/en/regexp.html)
+      // To allow case sensitive regexp query, it has to be binded to a `BINARY` type.
+      // If ignore case is not specified, search it as case sensitive.
+      if (!operatorValue.ignoreCase) {
+        clause = columnName + ' REGEXP BINARY ?';
+      }
 
-    if (operatorValue.multiline)
-      g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`m`}} flag');
+      if (operatorValue.ignoreCase)
+        g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`i`}} flag');
+      if (operatorValue.global)
+        g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`g`}} flag');
+
+      if (operatorValue.multiline)
+        g.warn('{{MySQL}} {{regex}} syntax does not respect the {{`m`}} flag');
+
+      return new ParameterizedSQL(clause,
+        [operatorValue.source]);
+    case 'nlmatch':
```

Comment:
I wonder why you didn't name it as `match` + the `search modifier` .  Remember that the operator in `match` but the rest are the modifiers so having the naming convention `function` + `modifier` would be better. 

`matchnl`, `matchqe`, `matchbool`,  `matchnlqe` & finally `match`.

Reply:
That does make more sense, I'll do this as well. I did'nt put much thought into the operator names, just followed the nlike, nilike pattern.